gmso/core/topology.py

```diff
@@ -965,6 +966,171 @@ def get_index(self, member):
+                            )
+                            for site in self.sites
+                        )
+                    except AttributeError:
+                        raise AttributeError(
+                            "The label {} is not in this gmso object".format(
+                                label
+                            )
+                        )
+                elif label == "positions" or label == "position":
+                    df["x"] = list(
+                        _return_float_for_unyt(
+                            getattr(site, "position")[0], unyts_bool
+                        )
+                        for site in self.sites
+                    )
+                    df["y"] = list(
+                        _return_float_for_unyt(
+                            getattr(site, "position")[1], unyts_bool
+                        )
+                        for site in self.sites
+                    )
+                    df["z"] = list(
+                        _return_float_for_unyt(
+                            getattr(site, "position")[2], unyts_bool
+                        )
+                        for site in self.sites
+                    )
+                elif label == "charge" or label == "charges":
+                    df["charge (e)"] = list(
+                        _return_float_for_unyt(
+                            site.charge
+                            / u.electron_charge
+                            * u.electron_charge.units,
+                            unyts_bool,
+                        )
+                        for site in self.sites
+                    )
+                else:
+                    try:
+                        df[label] = list(
+                            _return_float_for_unyt(
+                                getattr(site, label), unyts_bool
+                            )
+                            for site in self.sites
+                        )
+                    except AttributeError:
+                        raise AttributeError(
+                            "The label {} is not in this gmso object".format(
+                                label
+                            )
+                        )
+        elif parameter == "bonds":
+            df = self._pandas_from_parameters(
+                df, parameter=parameter, labels=labels, unyts_bool=unyts_bool
+            )
+            for i, param in enumerate(self.bonds[0].bond_type.parameters):
+                df[
+                    "Parameter "
+                    + str(i)
+                    + " ("
+                    + str(param)
+                    + "): "
+                    + str(self.bonds[0].bond_type.parameters[param].units)
+                ] = list(
+                    _return_float_for_unyt(
+                        bond.bond_type.parameters[param], unyts_bool
+                    )
+                    for bond in self.bonds
+                )
```

Comment:
Should this block of code go in the `_pandas_from_parameteres`?

Reply:
same for angles, dihedrals

Reply:
Based on above comments from you and @justinGilmer, I added new functions that pull this out of the main to_dataframes function!